### PR TITLE
Massive PHP rewrite to make coexist php7 and php5 at same time and etc.

### PR DESCRIPTION
**Pull request by vakartel**
the main idea: 
- php5 and php7 use near the same build schema.
- added trigger to switch system default php binary (a kind of update-alternatives)

### CODE REVIEW

**Comment by vakartel** [comment]
@andypost what do you think about such upgrade?
Any ideas are welcome

**Comment by andypost** [comment]
@vakartel Looks very promissing! Will review in details tomorrow.
The main problems:
- this should be single commit (please merge squash)
- commit message should be like https://github.com/alpinelinux/aports/pull/355

File: main/php5/php5-fpm.initd [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/sbin/openrc-run
+
+# If you want to run separate master process per pool, then create a symlink
+# to this runscript for each pool. In that mode, the php-fpm daemon is started
+# as nobody by default. You can override the user (and group) by declaring
+# variable "user" and optionally "group" in conf.d file, or in the $cfgfile
+# file (the former has precedence).
+
+command="/usr/sbin/php7-fpm"
```
Comment (andypost) [inline main/php5/php5-fpm.initd]: why php7 in php5 file?

File: community/php7/php7-module.conf [hunk 2]
```diff
@@ -1,4 +1,4 @@
-LoadModule php7_module modules/mod_php7.so
+LoadModule php7_module modules/libphp7.so
```
Comment (andypost) [inline community/php7/php7-module.conf]: any reason to rename?

File: community/php7/php7-fpm.logrotate [hunk 3]
```diff
@@ -8,6 +8,6 @@
 	dateext
 	sharedscripts
 	postrotate
-		/etc/init.d/php-fpm7 --ifstarted --quiet reopen
+		/etc/init.d/php7-fpm --ifstarted --quiet reopen
```
Comment (andypost) [inline community/php7/php7-fpm.logrotate]: why that changed?

File: community/php7/php7-module.conf (hunk 2 above)
Comment (vakartel) [inline community/php7/php7-module.conf]: in php5 it named libphp5. So I decided not to change it's name and stay it as is.

File: community/php7/php7-fpm.logrotate (hunk 3 above)
Comment (vakartel) [inline community/php7/php7-fpm.logrotate]: Firstly I remove suffixes at all. But then to implement coexistance with php5 I make suffixed 'php' part of name. 

It's mostly for install-scripts and trigger

**Comment by vakartel** [comment]
Don't know how to rename PR because it affects two independent packages.

@andypost , I'm not too familiar with git, what do you mean about merge squash?

**Comment by andypost** [comment]
@vakartel I mean to create new branch, squash commits into single commit here, delete old branch and rename new to old one. Details http://stackoverflow.com/questions/5308816/how-to-use-git-merge-squash

**Comment by vakartel** [comment]
todo:
- need change pear dir from /usr/share/php to /usr/share/pear for compatibity with existing pear-s.
- suffixed commands naming still in discussion...

then I will make two separate PR-s for php5 and php7 and close this one.

**Pull request closed without merging**